### ISSUE TITLE
Accessibility attributes are missing

### ISSUE THREAD

**Issue by mattesmohr** [vapor-community/HTMLKit#90]
The attributes for accessibility (https://www.w3.org/TR/wai-aria-1.1/) aren't implemented yet.

**Comment by mattesmohr** [comment]
For document reason:


Variant 1 | Variant 2 | Variant 3 | Variant 4 | Variant 5
-- | -- | -- | -- | --
.ariaActiveDescendant(String) |   | .aria(activedescendat: String) |   |  
.ariaAtomic(Bool) |   | .aria(atomic: Bool) | .atomic() | .isAtomic()
.ariaAutoComplete(Token) |   | .aria(autocomplete: Token) |   |  
.ariaBusy(Bool) |   | .aria(busy: Bool) | .busy() | .isBusy()
.ariaChecked(Token) |   | .aria(checked: Token) |   |  
.ariaColumnCount(Int) | .ariaColumn(count: Int) | .aria(column count: Int) |   |  
.ariaColumnIndex(Int) | .ariaColumn(index: Int) | .aria(columnindex: Int) |   |  
.ariaColumnSpan(Int) | .ariaColumn(span: Int) | .aria(columnspan: Int) |   |  
.ariaControls(String) |   | .aria(controls: String) |   |  
.ariaCurrent(Token) |   | .aria(current: Token) |   |  
.ariaDetails(String) |   | .aria(details: String) |   |  
.ariaDescribedBy(String) |   | .aria(described: String) |   |  
.ariaDisabled(Bool) |   | .aria(disabled: Bool) |   |  
.ariaDropEffect() |   | .aria(dropeffect:) |   |  
.ariaErrorMessage(String) |   | .aria(error: String) |   |  
.ariaExpanded(Bool) |   | .aria(expanded: Bool) |   |  
.ariaFlowTo(String) |   | .aria(flowto: String) |   |  
.ariaHasPopup(Bool) |   | .aria(popup: Bool) |   |  
.ariaHidden(Bool) |   | .aria(hidden: Bool) |   |  
.ariaInvalid(Bool) |   | .aria(invalid: Bool) |   |  
.ariaKeyShortCuts(String) |   | .aria(keyshortcuts: String) |   |  
.ariaLabel(String) |   | .aria(label: String) |   |  
.ariaLabeled(String) |   | .aria(labeled: String) |   |  
.ariaLevel(Int) |   | .aria(level: Int) |   |  
.ariaModal(Bool) |   | .aria(modal: Bool) |   |  
.ariaMultiline(Bool) |   | .aria(multiline: Bool) |   |  
.ariaMultiSelectable(Bool) |   | .aria(multiselect: Bool) |   |  
.ariaOrientation(Token) |   | .aria(orientation: Token) |   |  
.ariaOwns(String) |   | .aria(owns: String) |   |  
.ariaPlaceholder(String) |   | .aria(placeholder: String) |   |  
.ariaPositionInset(Int) |   | .aria(positioninset: Int) |   |  
.ariaPressed(Token) |   | .aria(pressed: Token) |   |  
.ariaReadyOnly(Bool) |   | .aria(readonly: Bool) |   |  
.ariaRelevant(Token) |   | .aria(relevant: Token) |   |  
.ariaRequired(Bool) |   | .aria(required: Bool) |   |  
.ariaRoleDescription(String) |   | .aria(roledescription: String) |   |  
.ariaRowCount(Int) | .ariaRow(count: Int) | .aria(rowcount: Int) |   |  
.ariaRowIndex(Int) | .ariaRow(index: Int) | .aria(rowindex: Int) |   |  
.ariaRowSpan(Int) | .ariaRow(span: Int) | .aria(rowspan: Int) |   |  
.ariaSelected(Token) |   | .aria(selected: Token) |   |  
.ariaSetSize(Int) |   | .aria(setsize: Int) |   |  
.ariaSort(Token) |   | .aria(sort: Token) |   |  
.ariaValueMaximum(Float) | .ariaValue(maximum: Float) | .aria(valuemaximum: Float) |   |  
.ariaValueMinimum(Float) | .ariaValue(minimum: Float) | .aria(valueminimum: Float) |   |  
.ariaValueNow(Float) | .ariaValue(now: Float) | .aria(valuenow: Float) |   |  
.ariaValueText(String) | .ariaValue(text: String) | .aria(valuetext: String) |   |

**Comment by mattesmohr** [comment]
Hey @MatsMoll, I think I need another opinion on this. I want to add accessibility attributes to the HTMLKit, since for ScreenReaders, its still a thing.

So I went through the docs and I hoped I can implement them, as I did with the event attributes lately.

```swift
Tag {
}
.on(event: .drag, "")
```

But the aria attributes have not the same value type, wich on the event attributes is always `String`. So I have to implement them differently. 

We usually do something like

```swift
Tag {
}
.camelCase()
```

right? So I listed all the variants one post above. I would like to hear you opinion on wich variant you like the most.

Variant 1 and Variant 2 is like we would normally do.
Variant 3 is more like the implementation of the event attributes. Seeing `.on` and `.aria` as one single attribute.
Variant 4 and 5 is like the first and the second but without the hint it is an aria-attribute.

I know its more like a question about "style", so don't rip my head off. :-D

```swift
Tag {
}
.id("tag)
.class("test test-style")
.ariaLabeled("Test")
.ariaModal(true)

Tag {
}
.id("tag)
.class("test test-style")
.aria(labeled: "Test")
.aria(modal: true)
```

**Comment by MatsMoll** [comment]
Hmm.. 🤔 

Personally do I like the `.aria(model: true)` syntax, as I find it clearer to separate the "generic" `aria` from the more "important" `model` or what ever it is. 

So I would maybe like the variant 3 with camel case in the attribute name ass well. E.g: `.aria(columnCount: Int)`. Or would that be a bit odd?

**Comment by mattesmohr** [comment]
> Hmm.. 🤔
> 
> Personally do I like the `.aria(model: true)` syntax, as I find it clearer to separate the "generic" `aria` from the more "important" `model` or what ever it is.
> 
> So I would maybe like the variant 3 with camel case in the attribute name ass well. E.g: `.aria(columnCount: Int)`. Or would that be a bit odd?

I will try it out and let you know. Thank you 👍

**Comment by mattesmohr** [comment]
For documentation reasons:

| Element           | Implicit Role    | Additional Roles                                              | Attributes                 |
|-------------------|------------------|---------------------------------------------------------------|----------------------------|
| Head              | -                | -                                                             | -                          |
| Title             | -                | -                                                             | -                          |
| Base              | -                | -                                                             | -                          |
| Link              | -                | -                                                             | -                          |
| Meta              | -                | -                                                             | -                          |
| Style             | -                | -                                                             | -                          |
| Html              | document         | -                                                             | -                          |
| Body              |                  |                                                               | global, aria-* (document)  |  
| Article           | article          | application, document, feed, main, none, presentation, region | global, aria-*             |
| Section           | region           |                                                               | global, aria-*             |
| Navigation        | navigation       | menu, menubar, tablist                                        | global, aria-*             |
| Aside             | complementary    | feed, none, note, presentation, region, search                | global, aria-*             |
| Heading1          | heading          | none, presentation, tab                                       | global, aria-*             |
| Heading2          | heading          | none, presentation, tab                                       | global, aria-*             |
| Heading3          | heading          | none, presentation, tab                                       | global, aria-*             |
| Heading4          | heading          | none, presentation, tab                                       | global, aria-*             |
| Heading5          | heading          | none, presentation, tab                                       | global, aria-*             |
| Heading6          | heading          | none, presentation, tab                                       | global, aria-*             |
| HeadingGroup      |                  | any                                                           | global, aria-*             |
| Header            |                  | group, none, presentation                                     | global, aria-*             |
| Footer            |                  | group, none, presentation                                     | global, aria-*             |
| Address           |                  |                                                               |                            |
| Paragraph         |                  | any                                                           | global, aria-*             |
| HorizontalRule    | separator        | none, presentation                                            | global, aria-* (separator) |
| PreformattedText  |                  | any                                                           | global, aria-*             |
| Blockquote        |                  | any                                                           | global, aria-*             |
| OrderedList       | list             | directory, group, listbox, menu, menubar, none, presentation  | global, aria-*             |
| UnorderedList     | list             | directory, group, listbox, menu, menubar, none, presentation  | global, aria-*             |
| ListItem          | listitem         | menuitem, menuitemcheckbox, menuitemradio, option, none,      | global, aria-*             |
| DescriptionList   |                  | group, list, presentation, none                               | global, aria-*             |
| TermName          | term             | listitem                                                      | global, aria-*             |
| TermDefinition    |                  |                                                               |                            |
| Figure            | figure           |                                                               | global, aria-*             |
| FigureCaption     |                  | group, presentation, none                                     | global, aria-*             |
| Main              | main             |                                                               | global, aria-* (main)      |
| Division          |                  | any                                                           | global, aria-*             |
| Anchor            |                  |                                                               |                            |
| Emphasize         |                  |                                                               |                            |
| Strong            |                  |                                                               |                            |
| Small             |                  |                                                               |                            |
| Cite              |                  |                                                               |                            |
| ShortQuote        |                  |                                                               |                            |
| Definition        |                  |                                                               |                            |
| Abbreviation      |                  |                                                               |                            |
| Ruby              |                  |                                                               |                            |
| RubyText          |                  |                                                               |                            |
| RubyPronunciation |                  |                                                               |                            |
| Data              |                  |                                                               |                            |
| Time              |                  |                                                               |                            |
| Code              |                  |                                                               |                            |
| Variable          |                  |                                                               |                            |
| SampleOutput      |                  |                                                               |                            |
| KeyboardInput     |                  |                                                               |                            |
| Subscript         |                  |                                                               |                            |
| Superscript       |                  |                                                               |                            |
| Italic            |                  |                                                               |                            |
| Bold              |                  |                                                               |                            |
| Underline         |                  |                                                               |                            |
| Mark              |                  |                                                               |                            |
| Bdi               |                  |                                                               |                            |
| Bdo               |                  |                                                               |                            |
| Span              |                  |                                                               |                            |
| LineBreak         |                  |                                                               |                            |
| WordBreak         |                  |                                                               |                            |
| InsertedText      |                  |                                                               |                            |
| DeletedText       |                  |                                                               |                            |
| Picture           |                  |                                                               |                            |
| Source            |                  |                                                               |                            |
| Image             |                  |                                                               |                            |
| InlineFrame       |                  |                                                               |                            |
| Embed             |                  |                                                               |                            |
| Object            |                  |                                                               |                            | 
| Parameter         |                  |                                                               |                            |
| Video             |                  |                                                               |                            | 
| Audio             |                  |                                                               |                            |
| Track             |                  |                                                               |                            |
| Map               |                  |                                                               |                            |
| Area              |                  |                                                               |                            |
| Table             |                  |                                                               |                            |
| Caption           |                  |                                                               |                            |
| ColumnGroup       |                  |                                                               |                            |
| Column            |                  |                                                               |                            |
| TableBody         |                  |                                                               |                            |
| TableHead         |                  |                                                               |                            |
| TableFoot         |                  |                                                               |                            |
| TableRow          |                  |                                                               |                            |
| DataCell          |                  |                                                               |                            |
| HeaderCell        |                  |                                                               |                            |
| Form              |                  |                                                               |                            |
| Label             |                  |                                                               |                            |
| Input             |                  |                                                               |                            |
| Button            |                  |                                                               |                            |
| DataList          |                  |                                                               |                            |
| OptionGroup       |                  |                                                               |                            |
| Option            |                  |                                                               |                            |
| TextArea          |                  |                                                               |                            |
| Output            |                  |                                                               |                            |
| Progress          |                  |                                                               |                            |
| Meter             |                  |                                                               |                            |
| Fieldset          |                  |                                                               |                            |
| Legend            |                  |                                                               |                            |
| Details           |                  |                                                               |                            |
| Summary           |                  |                                                               |                            |
| Dialog            |                  |                                                               |                            |
| Script            |                  |                                                               |                            |
| NoScript          |                  |                                                               |                            |
| Template          |                  |                                                               |                            |
| Canvas            |                  |                                                               |                            |
| Vector            |                  |                                                               |                            |
| Circle            |                  |                                                               |                            |
| Rectangle         |                  |                                                               |                            |
| Ellipse           |                  |                                                               |                            |
| Line              |                  |                                                               |                            |
| Polygon           |                  |                                                               |                            |
| Polyline          |                  |                                                               |                            |
| Path              |                  |                                                               |                            |
| Use               |                  |                                                               |                            |
| Select            |                  |                                                               |                            |
